tests/integration/test_next.py

```diff
@@ -0,0 +1,45 @@
+import pytest
+
+from rebirthdb.errors import RqlCursorEmpty
+from tests.helpers import IntegrationTestCaseBase, INTEGRATION_TEST_DB
+
+
+@pytest.mark.integration
+class TestCursorNext(IntegrationTestCaseBase):
+    def setup_method(self):
+        super(TestCursorNext, self).setup_method()
+        self.table_name = 'cursor_next'
+        self.r.table_create(self.table_name).run(self.conn)
+        self.documents = [
+            {'id': 1, 'name': 'Testing Cursor/Next 1'},
+            {'id': 2, 'name': 'Testing Cursor/Next 2'},
+        ]
+
+    def teardown_method(self):
+        self.r.table_drop(self.table_name).run(self.conn)
+        super(TestCursorNext, self).teardown_method()
+
+    def test_get_next_document(self):
+        self.r.table(self.table_name).insert(self.documents).run(self.conn)
+
+        cursor = self.r.table(self.table_name).run(self.conn)
+
+        for document in reversed(self.documents):
+            assert document == cursor.next()
```

Comment:
This order has no guarantees. A possible fix would be to use a temporary set copy of the documents and test that they all come back in some order.

Reply:
As I can see, the order is fixed, due to the “next returns the next document”.

Reply:
Hm, that it right. Tomorrow I will review the test